Use esp-azure as an esp-idf project component (CA-42)

fgiancane8 opened the pull request:
Hi,
I tried to include `esp-azure` as a component (cloning it in `components` folder of a project). Unfortunately, the build failed. You need to explicitly set the extra component path in `CMakeLists.txt`.

With these patches, it is now possible to just download it in `components` and have it automatically linked and built.

Commit:
CMakeLists.txt: move to root directory

Move CMakeLists.txt in root directory. This allows the repository to be
cloned inside the `components` directory of an `esp-idf` project and
automatically build and link to it the Azure SDK.

This is also the behavior included in `component.mk` Makefile.

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

Commit:
examples/: update CMakeLists.txt for `examples` projects

Projects now do not require to explictly declare the `esp-azure`
component as an extra components. Just copy the examples and drop (or
link) the `esp-azure` folder in `components` subdirectory.

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

Commit:
README.md: add a simple README file for examples

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/espressif/esp-azure?pullRequest=78) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/espressif/esp-azure?pullRequest=78) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/espressif/esp-azure?pullRequest=78) it.</sub>

Commit:
azure-iot-sdk-c: update to version 1.3.7

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

Commit:
CMakeLists.txt: update source file list for Azure SDK

Moving from 1.2.x to 1.3.x version in Azure SDK required some changes to
the built source code list.

Also, the source folders for Azure SDK were restructured.

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

Commit:
port/inc/tlsio_pal.h: update #include files

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

Commit:
port/src/agenttime_esp.c: time functions are now built-in in Azure SDK

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

Commit:
port/src/platform_esp.c: update platform identification function
signature

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

Commit:
port/src/tlsio_esp_tls.c: update test cases return symbol

__FAILURE__ was substituted by MU_FAILURE. Apply required changes.

Signed-off-by: Francesco Giancane <francesco.giancane@accenture.com>

fgiancane8 commented:
Sorry, I created the PR from the wrong branch. Reopening it.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/espressif/esp-azure?pullRequest=78) <br/>All committers have signed the CLA.
